stdlib/2and3/contextlib.pyi

```diff
@@ -64,3 +64,29 @@ if sys.version_info >= (3,):
         def pop_all(self) -> ExitStack: ...
         def close(self) -> None: ...
         def __enter__(self: _U) -> _U: ...
+
+if sys.version_info >= (3, 7):
+    from typing import Awaitable
+
+    _U = TypeVar('_U', bound='AsyncExitStack')
+
+    _ExitCoroFunc = Callable[[Optional[Type[BaseException]],
+                              Optional[BaseException],
+                              Optional[TracebackType]], Awaitable[bool]]
+    _CallbackCoroFunc = Callable[..., Awaitable[None]]
+    _ACM_EF = TypeVar('_ACM_EF', AsyncContextManager, _ExitCoroFunc)
+
+    class AsyncExitStack(AsyncContextManager[AsyncExitStack]):
+        def __init__(self) -> None: ...
+        def enter_context(self, cm: ContextManager[_T]) -> _T: ...
+        def enter_async_context(self, cm: AsyncContextManager[_T]) -> Awaitable[_T]: ...
+        def push(self, exit: _CM_EF) -> _CM_EF: ...
+        def push_async_exit(self, exit: _ACM_EF) -> _ACM_EF: ...
+        def callback(self, callback: Callable[..., None],
+                     *args: Any, **kwds: Any) -> Callable[..., None]: ...
+        def push_async_callback(self, callback: _CallbackCoroFunc,
+                                *args: Any, **kwds: Any) -> _CallbackCoroFunc: ...
+        def pop_all(self) -> ExitStack: ...
+        def aclose(self) -> Awaitable[None]: ...
+        def __enter__(self: _U) -> _U: ...
+        def __aenter__(self: _U) -> Awaitable[_U]: ...
```

Comment:
Do you not need `__exit__` and `__aexit__`? I suppose the base class provides those.

Reply:
Base class implementations should be enough.

Reply:
Actually, that's true for `__aexit__` but not `__exit__`.

Reply:
Why not? Definitions in typing.pyi seem legit ([`__exit__`](https://github.com/python/typeshed/blob/71c2a94cb896b2b121b53ab66777a9819265f00e/stdlib/3/typing.pyi#L317-L319) / [`__aexit__`](https://github.com/python/typeshed/blob/71c2a94cb896b2b121b53ab66777a9819265f00e/stdlib/3/typing.pyi#L325-L327))

Reply:
Or do you mean that's because `AsyncContextManager` does not inherit from `ContextManager`?

Reply:
Oh, nvm. There is not `__enter__` for `AsyncExitStack`.